README.md

```diff
@@ -280,6 +282,44 @@ config.loadFile('config.toml');
 If no supported extension is detected, `loadFile` will fallback to using the
 default json5 parser for backward compatibility.
 
+## Pluggable external providers support
+
+Convict supports loading values synchronously from external providers -- any key/value store, such as the AWS Parameter store -- can be used to bootstrap configuration. 
+
+This is accomplished by passing a function that (synchronously) accepts a key and returns a value into `convict.configureProvider()`. 
+
+For instance:
+
+```javascript
+const convict = require("node-convict");
+
+let provider = function(providerPath) {
+  let out = {
+    "/service/ip": "10.0.1.101",
+    port: "8080"
+  };
+
+  return out[providerPath];
+};
+
+convict.configureProvider(provider);
```

Comment:
Why ????

You can do : 

```
convict({/*...*/}).load({
    "/service/ip": "10.0.1.101",
    port: "8080"
}).validate();
```

It is exactly the same thing because we can only provide one function not several.

Reply:
It appears from the code that provider's operate from a path, which would align with my anticipated behavior for this kind of feature. See `test/cases/provider_multi.out` and `test/cases/provider_precedent.js`. 

A real world example would be to have config values such as `s3.username` or `vault.db_password`.

Reply:
@A-312 what is confusing about my PR description, readme and extensive tests? This allows Convict to pull values from external providers by injecting a custom, user-provided function. 

An example is here: https://github.com/HBKEngineering/convict-provider-awsSsm/blob/master/index.js, which pulls values from AWS Parameter Store. So, no, you cannot simply do:

```
convict({/*...*/}).load({
    "/service/ip": "10.0.1.101",
    port: "8080"
}).validate();```

You can imagine writing a similar function for S3, Vault... any key/value store.

As I mentioned in the PR, I understand if this is not the kind of functionality you want to add to `node-convict` but our team has found it extremely useful and it's running in all of our services and has been for years, so I figured I would submit a PR and contribute back to this extremely helpful tool rather than selfishly maintain our own fork. 

I'm not really sure the rudeness from your end is justified and it makes me somewhat sad to see what seems to be some amount of aggression from a contributor Mozilla-operated project, but if you are simply confused I'm happy to clarify any questions and comments you have.
 
Also, of course it has merge conflicts. I submitted it almost a year ago, and would've appreciated the feedback then. I figured the project had simply been abandoned, and had planned on helping contribute to it until the radio silence.

Reply:
I read all the new line of the readme but I don't understand the difference between `.load(function() { return { a: 1, b: 2}; })` and provider. Why does this need a new function? This seems a lot of effort but I don't understand so many changes for one thing.

---------

> Also, of course it has merge conflicts. I submitted it almost a year ago, and would've appreciated the feedback then. I figured the project had simply been abandoned, and had planned on helping contribute to it until the radio silence.

ping @madarche

Reply:
If I can recall correctly, the issue had to do something with coercion. I'll take a look at it in more depth tomorrow.